src/include/self_driving/planning/planning_context.h

```diff
@@ -172,6 +241,11 @@ class PlanningContext {
   std::set<catalog::db_oid_t> db_oids_;
   std::unordered_map<catalog::db_oid_t, common::ManagedPointer<transaction::TransactionContext>> db_oid_to_txn_;
   std::unordered_map<catalog::db_oid_t, std::unique_ptr<catalog::CatalogAccessor>> db_oid_to_accessor_;
+  // <ou, <feature, cached label>>
+  std::map<ExecutionOperatingUnitType, std::map<std::vector<int>, std::unique_ptr<std::vector<double>>>>
```

Comment:
I think sometimes we need to return a pointer/reference of the cached value to avoid copying. I wasn't sure whether it's safe to return a reference of a non-pointer value. Because if the map is modified, that reference may become invalid. If we use unique pointer, then I think it should always be safe to return the pointer value because that's inside the unique pointer class.